espnet2/spk/encoder/rawnet3_encoder.py

```diff
@@ -0,0 +1,94 @@
+# Copyright 2023 Jee-weon Jung
+# Apache 2.0
+
+"""RawNet3 Encoder"""
+
+from typing import List, Optional, Tuple
+
+import torch
+import torch.nn as nn
+from asteroid_filterbanks import Encoder, ParamSincFB
+from typeguard import check_argument_types
+
+from espnet2.asr.encoder.abs_encoder import AbsEncoder
+from espnet2.spk.layers.RawNetBasicBlock import Bottle2neck, PreEmphasis
+
+
+class RawNet3Encoder(AbsEncoder):
+    """
+    RawNet3 encoder. Extracts frame-level RawNet embeddings from raw waveform.
+    paper: J. Jung et al., "Pushing the limits of raw waveform speaker
+        recognition", in Proc. INTERSPEECH, 2022.
+
+    Args:
+        block: type of encoder block class to use.
+        model_scale: scale value of the Res2Net architecture.
+        output_size: dimensionality of the hidden representation.
+        sinc_stride: stride size of the first sinc-conv layer where it decides
+            the compression rate (Hz).
+    """
+
+    def __init__(
+        self,
+        block=Bottle2neck,
+        model_scale: int = 8,
+        output_size: int = 1024,
+        sinc_stride: int = 16,
+        **kwargs,
+    ):
+        assert check_argument_types()
+        super().__init__()
+        self._output_size = output_size
+
+        self.waveform_process = nn.Sequential(
+            PreEmphasis(), nn.InstanceNorm1d(1, eps=1e-4, affine=True)
+        )
+        self.conv = Encoder(ParamSincFB(output_size // 4, 251, stride=sinc_stride))
+        self.relu = nn.ReLU()
+
+        self.layer1 = block(
+            output_size // 4,
+            output_size,
+            kernel_size=3,
+            dilation=2,
+            scale=model_scale,
+            pool=5,
+        )
+        self.layer2 = block(
+            output_size,
+            output_size,
+            kernel_size=3,
+            dilation=3,
+            scale=model_scale,
+            pool=3,
+        )
+        self.layer3 = block(
+            output_size, output_size, kernel_size=3, dilation=4, scale=model_scale
+        )
+        self.layer4 = nn.Conv1d(3 * output_size, 1536, kernel_size=1)
```

Comment:
Why is the output size fixed as 1536?

Reply:
thanks for pointing this out! I will commit a fixed version.